### PR TITLE
skip inactive history entries when generating history context

### PR DESCRIPTION
**Pull request by vladak**
This change fixes diff link generation for histories that contain inactive history entries.

### CODE REVIEW

**Review by ginoaugustine — APPROVED** [review]
LGTM

**Comment by vladak** [comment]
> LGTM

Actually, I found more elegant solution - filter the inactive history entries up front.

File: opengrok-indexer/src/test/java/org/opengrok/indexer/search/context/HistoryContextTest.java [hunk 1]
```diff
@@ -190,4 +198,37 @@ void testWriteMatch() throws IOException {
...
                 sb.toString());
     }
+
+    /**
+     * Test that inactive history entries are skipped when generating history context.
+     */
+    @Test
+    void testGetHistoryContextVsInactiveHistoryEntries() {
+        Set<String> filePaths = Set.of(File.separator + Paths.get("teamware", "foo.c"));
+        History history = new History(List.of(
```
Comment (vladak) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/search/context/HistoryContextTest.java:208]: It would be nice to test some corner cases, like all history entries are inactive or if the next entry is not present etc.,  however not this PR.
